### PR TITLE
Better stats registry

### PR DESCRIPTION
**Pull request by gibsong**
This PR is a dependency to https://github.com/ehcache/ehcache3/pull/1390

### CODE REVIEW

**Comment by terracotta-org** [comment]
Can one of the admins verify this patch?

File: src/main/java/org/terracotta/statistics/extended/SamplingSupport.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+package org.terracotta.statistics.extended;
+
+/**
+ * @author Ludovic Orban
+ */
+public interface SamplingSupport {
+
+  boolean expire(long expiry);
+
+  void setAlwaysOn(boolean enable);
```
Comment (mathieucarbou) [inline src/main/java/org/terracotta/statistics/extended/SamplingSupport.java]: `setAlwaysOn ` is the method that would avoid a stats to stop being computed, used for testing purposes, right ?

Comment (gibsong) [inline src/main/java/org/terracotta/statistics/extended/SamplingSupport.java]: shoot, i actually am not sure.  Ludovic wrote that code and I haven't seen it before.  I can look into it when I add in more stats.

**Comment by ljacomet** [comment]
test this please

**Comment by anthonydahanne** [comment]
sorry I missed the notification about this PR.
LGTM, @mathieucarbou good for you too ?

**Comment by mathieucarbou** [comment]
@anthonydahanne : yes! we'll just need to know the answer for my question, because I guess it will help the testing also for us :-)

**Pull request merged**